app/gui/src/controller/searcher.rs

```diff
@@ -1673,12 +1705,27 @@ pub mod test {
 
     #[wasm_bindgen_test]
     fn loading_components() {
+        // Prepare a sample component group to be returned by a mock Language Server client.
+        let exported_component_name =
+            crate::test::mock::data::module_qualified_name().to_string() + ".testFunction1";
+        let exported_component =
+            language_server::LibraryComponent { name: exported_component_name, shortcut: None };
+        let sample_ls_component_group = language_server::LibraryComponentGroup {
+            library: "".to_string(),
+            name:    "Test Group 1".to_string(),
+            color:   None,
+            icon:    None,
+            exports: vec![exported_component],
+        };
+        // Create a test fixture with mocked Engine responses.
         let Fixture { mut test, searcher, entry1, entry9, .. } =
             Fixture::new_custom(|data, client| {
                 // Entry with id 99999 does not exist, so only two actions from suggestions db
                 // should be displayed in searcher.
                 data.expect_completion(client, None, None, &[1, 99999, 9]);
+                data.graph.ctx.component_groups = vec![sample_ls_component_group];
             });
+        // Verify the contents of a components list loaded by the Searcher.
```

Comment:
What does this comment mean? How `reload_list` is used to "verify the contents"?